services/src/main/java/edu/unc/lib/dl/cdr/services/processing/RunEnhancementsService.java

```diff
@@ -110,11 +115,12 @@ public void setAclService(AccessControlService aclService) {
     private void createMessage(BriefObjectMetadata metadata, String username, Boolean force) {
         PID pid = metadata.getPid();
         Datastream originalDs = metadata.getDatastreamObject(ORIGINAL_FILE.getId());
-        if (originalDs == null) {
+        Datastream hasThumbnail = metadata.getDatastreamObject(THUMBNAIL_SMALL.getId());
```

Comment:
Hmm, I would rather leave it up to the enhancement processors to decide if anything needs to be done to an object rather than do it here, which we would need to remember to update in the future if more types of derivatives are added for non-files. So I'd suggest removing this and the conditional at 119. The shortcut to jump to the original ds is probably fine to keep.